### PR TITLE
Refactor dataset to use pandas and cleaner setup.

### PR DESCRIPTION
**Pull request by davidgasquez**
Heya @rufuspollock!

Since this seems to be the most starred package and also broken, I thought I would update/improve it after having spent some time with Frictionless and other data package managers.

It changes lots of things so please push back on anything that doesn't make sense.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "name": "Core Dataset",
+    "image": "mcr.microsoft.com/devcontainers/python:3.11",
+    "features": {
+        "ghcr.io/stuartleeks/dev-container-features/shell-history:0": {}
+    }
+}
```
Comment (davidgasquez) [inline .devcontainer/devcontainer.json:7]: With this, you can run the pipeline in your browser using [GitHub Codespaces](https://github.com/codespaces).

A simple way to make future contributors life easier as getting a development environment is one click away.

File: .github/workflows/actions.yml [hunk 2]
```diff
@@ -1,48 +1,30 @@
...
   push:
-    branches:
-      - master
-  #  2023-03-08 disable cron until working again
-  # schedule:
-  #  - cron:  '0 1 * * *'
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]
+  workflow_dispatch:
```
Comment (davidgasquez) [inline .github/workflows/actions.yml:6]: Make it possible to trigger the workflow from the UI.

File: datapackage.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        - name: GICS Sub-Industry
+          type: string
+        - name: Headquarters Location
+          type: string
+        - name: Date added
+          type: string
+        - name: CIK
+          type: string
+        - name: Founded
+          type: string
```
Comment (davidgasquez) [inline datapackage.yaml:30]: Moved to YAML as it makes it more readable and I is what modern Frictionless data packages are using around GitHub.

File: scripts/requirements.txt [hunk 4]
```diff
@@ -1 +1,3 @@
-beautifulsoup4
+pandas
```
Comment (rufuspollock) [inline scripts/requirements.txt:1]: I get it! Though pandas is so heavy duty (you have to install numpy right ...). I wonder if we can get away with something more lightweight.

Let's live with it for now!

**Review by rufuspollock — APPROVED** [review]

File: scripts/requirements.txt (hunk 4 above)
Comment (davidgasquez) [reply scripts/requirements.txt:1]: Agree. In this case I think is heavyweight for the computers but easy for humans to understand.

**Pull request merged**